### ISSUE TITLE
More authentication options

### ISSUE THREAD

**Issue by JKrag** [Praqma/Praqmatic-Automated-Changelog#55]

**Comment by buep** [comment]
Please elaborate.

**Comment by buep** [comment]
Authentication types we can chose support?

* basic auth (already supported)
* no-auth
* ssh key pair
* token

**Comment by MadsNielsen** [comment]
no-auth is easy to implmement, we can just add a check to see if `usr` and `pwd` are defined. That would make them optional attributes.

**Comment by buep** [comment]
If you need some more authentication options, please create and groom a specific issue for that specific auth option.... link to this. Then I can prioritize it.

**Comment by buep** [comment]
Backlog grooming: We have many options, but not:

* cookie based/session based

We have:

* no auth (http/https)
* basic (http/https)
